src/Nethermind/Nethermind.Consensus/Scheduler/BackgroundTaskScheduler.cs

```diff
@@ -158,51 +170,37 @@ private async Task StartChannel()
         }
     }
 
-    public void ScheduleTask<TReq>(TReq request, Func<TReq, CancellationToken, Task> fulfillFunc, TimeSpan? timeout = null)
+    public bool TryScheduleTask<TReq>(TReq request, Func<TReq, CancellationToken, Task> fulfillFunc, TimeSpan? timeout = null)
     {
-        timeout ??= DefaultTimeout;
-        DateTimeOffset deadline = DateTimeOffset.UtcNow + timeout.Value;
-
         IActivity activity = new Activity<TReq>
         {
-            Deadline = deadline,
+            Deadline = DateTimeOffset.UtcNow + (timeout ?? DefaultTimeout),
             Request = request,
             FulfillFunc = fulfillFunc,
         };
 
         Evm.Metrics.IncrementTotalBackgroundTasksQueued();
 
         bool success = false;
-        lock (_queueLock)
+
+        if (Interlocked.Increment(ref _queueCount) < _capacity)
         {
-            if (_queueCount + 1 < _capacity)
+            success = _taskQueue.Writer.TryWrite(activity);
+            if (success)
             {
-                success = _taskQueue.Writer.TryWrite(activity);
-                if (success)
-                {
-                    Interlocked.Increment(ref _queueCount);
-                }
+                UpdateQueueCount();
+            }
+            else
+            {
+                Interlocked.Decrement(ref _queueCount);
+                request.TryDispose();
```

Comment:
Add `_logger.Warn` that queue is full; maybe with timestamp check so min time between warns is 10sec rather than doing it every millisecond (as should never happen, so we really need to know if it is)